processor/k8sprocessor/kube/client.go

```diff
@@ -235,6 +236,30 @@ func (c *WatchClient) extractPodAttributes(pod *api_v1.Pod) map[string]string {
 			tags[r.Name] = c.extractField(v, r)
 		}
 	}
+
+	if c.Rules.HostName {
+		// Basing on v1.17 Kubernetes docs, when a hostname is specified,
+		// it takes precedence over the associated metadata name, see:
+		// https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields
+		hostname := pod.Spec.Hostname
+		if hostname == "" {
+			hostname = pod.Name
+		}
+		tags[conventions.AttributeHostName] = hostname
```

Comment:
I'm not sure if it's safe to reuse `host.name` attribute here. It can be easily confused with hostname of the node where the pod is running on. What is the use case? Do we have to reuse `host.name` attribute? if not, we might want to use another attribute with a more specific name like `k8s.pod.hostname`. what do you think?

cc @tigrannajaryan @jrcamp

Reply:
@dmitryax+1 on `k8s.pod.hostname`(I think there are several attributes that could be easily added to semantic conventions)

Reply:
Agreed @dmitryax.

We'll need to add it to https://github.com/open-telemetry/opentelemetry-collector/blob/master/translator/conventions/opentelemetry.go.

@pmm-sumo mentioned he'd like to contribute that so let's wait for that and use it here.

Reply:
I have started with opening a PR to specs (including two other fields): https://github.com/open-telemetry/opentelemetry-specification/pull/1072

Reply:
There's a discussion under the PR. I am wondering if maybe `host.hostname` is a better fit here. What do you think?

Reply:
@pmm-sumo the `host.hostname` was recently removed from the conventions https://github.com/open-telemetry/opentelemetry-specification/pull/838

Reply:
So to recap, we had a discussion under the specification PR. Since `hostname` is ambiguous in this setting (containers host names vs the machine where it runs host name), it is unfortunately not a simple problem to solve. Additionally, it is present on all containerised environments and need to be addressed in a generic way. 

To avoid taking focus out of the GA, the suggestion is to skip adding this feature for now and go back to it later, with a solid discussion involving several parties.

Reply:
Also, most of the time, the node hostname will reside in `k8s.node.name` and pod hostname in `k8s.pod.name` unless they are overwritten (which I believe is not a very common scenario).